schedule/views.py

```diff
@@ -9,108 +10,218 @@
 from .serializers import UserSerializer, ScheduleSerializer
 
 
+def logged_in(request):
+    if 'login' in request.session.keys():
+        if request.session['login'] == "Yes":
+            return True
+    return False
+
+
 @api_view(['GET', 'POST'])
-def list_users(request, format = None):
+def list_users(request, format=None):
     if request.method == 'GET':
-        users = Users.objects.all().filter(verified=True)
+        if not logged_in(request):
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+        users = Users.objects.all().filter(verified=True, deleted_flag=False)
         serializer = UserSerializer(users, many=True)
-        responses = serializer.data
-        for response in responses:
-            del response['password']
-            del response['verified']
-        return Response(responses)
+        responses = {}
+        try:
+            responses['data'] = serializer.data
+            responses['status'] = {'success': True,
+                                   'message': "Data Retrieved"}
+            for response in responses['data']:
+                del response['password']
+                del response['verified']
+                del response['deleted_flag']
+            return Response(responses)
+        except:
+            responses['status'] = {'success': False,
+                                   'error': "Data could not be retrieved"}
+            return Response(responses, status=status.HTTP_400_BAD_REQUEST)
 
     elif request.method == 'POST':
+        response = {}
         data = JSONParser().parse(request)
+        data['password'] = hashlib.sha1(data['password']).hexdigest()
```

Comment:
The database config I think by default has blank = false
Serializer should catch error